src/test/java/nl/knaw/dans/layerstore/LayerDatabaseAddDirectoryTest.java

```diff
@@ -163,6 +165,42 @@ public void should_add_item_records_even_if_directories_already_exist_in_another
             );
     }
 
+    @Test
+    public void should_find_generatedId_in_db() {
+        var e = assertThrows(OptimisticLockException.class, () ->
+                daoTestExtension.inTransaction(() -> {
+                    var record = ItemRecord.builder()
+                            .layerId(1L)
+                            .path("app")
+                            .type(Item.Type.Directory)
+                            .generatedId(21L)
+                            .build();
+                    dao.saveRecords(record);
+                })
+        );
+        AssertionsForClassTypes.assertThat(e.getMessage()).startsWith("Batch update returned unexpected row count from update [0]");
+    }
+    @Test
+    public void should_update_path() {
+        addToDb(1L, "james", Item.Type.Directory);
+        var r = dao.getRecordsByPath("james").get(0);
+        daoTestExtension.inTransaction(() -> {
+            r.setPath("bond");
+            dao.saveRecords(r);
+         });
+        assertThat(dao.getAllRecords().toList())
+                .usingRecursiveFieldByFieldElementComparator()
```

Comment:
Leave out the `.usingRecursiveFieldByFieldElementComparator()` if you are not ignoring the generatedId field. It serves no purpose now.